RyEngine/src/Core/Input/Mouse.cpp

```diff
@@ -15,9 +14,23 @@ namespace RE
 	{
 		ASSERT(eventType == EventType::RE_MOUSE_DOWN || eventType == EventType::RE_MOUSE_UP || eventType == EventType::RE_MOUSE_MOVE || eventType == EventType::RE_MOUSE_SCROLL);
 
+		if (_mEventCallbacks.find(eventType) != _mEventCallbacks.end())
+		{
+			RE_CORE_WARN("Mouse.SetMouseEventHandler: Callback already exists for type {0}", eventType);
+			return;
+		}
+
 		_mEventCallbacks[eventType].push_back(callback);
 	}
 
+	void Mouse::RemoveMouseEventHandler(EventType eventType)
+	{
+		if (_mEventCallbacks.find(eventType) != _mEventCallbacks.end()) 
+		{
+			_mEventCallbacks.erase(eventType);
```

Comment:
Same situation here. _mEventCallbacks[eventType] is a list of callbacks potentially added by several different systems. I don't think they should all be removed at once.